crawler.py

```diff
@@ -0,0 +1,91 @@
+""" Async Web Crawler """
+
+import asyncio
+import html
+import requests
+from re import findall, compile
+from requests_html import AsyncHTMLSession
+from typing import List
+
+
+async def get_html_content(url: str) -> AsyncHTMLSession.get:
+    """
+    Return an html content of a given url
+
+    :param url: A web url.
+    :return: Html content of a website.
+    """
+
+    session = AsyncHTMLSession()
+    response = await session.get(url)
+    return response
+
+
+async def get_all_links(url: str) -> List[str]:
+    """
+    Find all links in a given url
+
+    :param url: A web url.
+    :return: A list of all links found.
+    """
+
+    response = await get_html_content(url)
+    links = response.html.absolute_links
+    return links
+
+
+class web_crawler:
```

Comment:
Move the class to another module. I would suggest to `crawler.py`. Separating to modules doesn't cost shekels.